### ISSUE TITLE
Que 2.1.0: fails to reschedule self job

### ISSUE THREAD

**Issue by seansfkelley** [hlascelles/que-scheduler#360]
# Description

I see in #332 that Que 2.x support is experimental, so I experimented! It appears that some ID has changed from/to a UUID/integer in the schema, so the reschedule-self job is failing validation when it tries to insert a replacement job for itself.

It does appear to be successfully scheduling the non-self jobs. I did a smoke test that just prints:

```
19:07:01 worker.1 | "sanity check"
19:07:06 worker.1 | "sanity check"
19:07:26 worker.1 | "sanity check"
19:08:52 worker.1 | "sanity check"
19:13:13 worker.1 | "sanity check"
19:23:42 worker.1 | "sanity check"
```

This test job was scheduled with `7 * * * *`. The timestamps suggest that it's being successfully scheduled (and then run) every time the scheduler job runs, but that the scheduler job itself is backing off due to the failures.

# Versions

ruby version: 3.1.2
que version: 2.1.0 (repros also with 2.0.0, which has a different job schema than 2.1.0 but I didn't check how)
que-scheduler version: 4.3.1 (forked to allow que 2.x in the gemspec; no other changes)
postgres version: 14.4

# Messages

Please provide all error messages and stacktraces.

error:

```
PG::InvalidTextRepresentation: ERROR:  invalid input syntax for type bigint: "12ce6a4b-6bfa-4c42-924c-9cb17e2004f3"
CONTEXT:  unnamed portal parameter $6 = '...'
```

backtrace:

```
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection.rb:65:in `exec_params'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection.rb:65:in `block in execute'
(eval):9:in `run_sql_middleware'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection.rb:60:in `execute'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection_pool.rb:57:in `block in execute'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection_pool.rb:39:in `block in checkout'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:13:in `block (2 levels) in checkout'
/Users/seankelley/.rvm/gems/ext-3.1.2/gems/activerecord-7.0.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:215:in `with_connection'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:12:in `block in checkout'
/Users/seankelley/.rvm/gems/ext-3.1.2/gems/activesupport-7.0.2.3/lib/active_support/execution_wrapper.rb:88:in `wrap'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:23:in `wrap_in_rails_executor'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:11:in `checkout'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection_pool.rb:18:in `checkout'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection_pool.rb:57:in `execute'
/opt/homebrew/Cellar/ruby/3.1.2/lib/ruby/3.1.0/forwardable.rb:238:in `execute'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/version_support.rb:47:in `execute'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/audit.rb:29:in `block in append'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/audit.rb:28:in `each'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/audit.rb:28:in `append'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/scheduler_job.rb:64:in `enqueue_self_again'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/scheduler_job.rb:32:in `block in run'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/db.rb:23:in `block in transaction'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/utils/transactions.rb:18:in `block in transaction'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection_pool.rb:39:in `block in checkout'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:13:in `block (2 levels) in checkout'
/Users/seankelley/.rvm/gems/ext-3.1.2/gems/activerecord-7.0.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:215:in `with_connection'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:12:in `block in checkout'
/Users/seankelley/.rvm/gems/ext-3.1.2/gems/activesupport-7.0.2.3/lib/active_support/execution_wrapper.rb:92:in `wrap'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:23:in `wrap_in_rails_executor'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:11:in `checkout'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/connection_pool.rb:18:in `checkout'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/utils/transactions.rb:12:in `transaction'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/db.rb:23:in `call'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/db.rb:23:in `transaction'
/opt/homebrew/lib/ruby/gems/3.1.0/bundler/gems/que-scheduler-1c9437fe1ff6/lib/que/scheduler/scheduler_job.rb:23:in `run'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/job_methods.rb:51:in `_run'
<internal:kernel>:90:in `tap'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/worker.rb:103:in `block in work_job'
(eval):11:in `block in run_job_middleware'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/utils/middleware.rb:41:in `invoke_middleware'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/utils/middleware.rb:37:in `block in invoke_middleware'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/active_record/connection.rb:33:in `call'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/utils/middleware.rb:36:in `invoke_middleware'
(eval):11:in `run_job_middleware'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/worker.rb:103:in `work_job'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/worker.rb:76:in `work_loop'
/opt/homebrew/lib/ruby/gems/3.1.0/gems/que-2.1.0/lib/que/worker.rb:46:in `block in initialize'
```

**Comment by seansfkelley** [comment]
After much debugging and banging my head against a wall, it looks like somehow I ended up in a configuration where I was using the Rails default Async queue but also que-scheduler at the same time, so it was receiving UUIDs from the Async queue (since it's not backed by a database, I guess it prefers UUIDs to avoid any potential collisions). Correcting the configuration to actually use Que seems to resolve the issue!

**Comment by hlascelles** [comment]
Ah nice... And I was about to thank you for the perfect bug report too! (all the versions, the background, the stack trace etc...)

Glad you've got it sorted. I'll take this as a good prompt to add the new version to the testing matrix. 👍

**Comment by seansfkelley** [comment]
> ...add the new version to the testing matrix.

Does that mean it'll permit using Que 2.x in the gemspec? That would be super because I had to fork que-scheduler just to relax the requirement cause (AFAIK) bundler doesn't let you force version overrides. :/

**Comment by hlascelles** [comment]
Good spot... I will change that now :+1:

**Comment by hlascelles** [comment]
It's coming... I'm adding it to this PR of dependabot's to allow all up to 3.0. https://github.com/hlascelles/que-scheduler/pull/363

It looks like there is now a minimum requirement of Postgres 9.6.0, but I think that is fine.

If that passes I'll do a release. Thanks for your patience!

**Comment by hlascelles** [comment]
Released 4.4.0 with que support to < 3.0.0: https://rubygems.org/gems/que-scheduler/versions/4.4.0
